src/main/java/org/cactoos/text/Abbreviated.java

```diff
@@ -80,32 +82,24 @@ public Abbreviated(final String text, final int max) {
      * Ctor.
      * @param text The Text
      * @param max Max width of the result string
-     * @todo #1287:30min Introduce `text.Flatten` that takes a `Scalar` of `Text`
-     *  and call `value()` then `asString()` on it. Add some tests for it (including
-     *  for `equals`). Then replace the code below by a composition of `text.Flatten`
-     *  and `scalar.Ternary`. Then do the same for `PrefixOf` and `SuffixOf` using
-     *  `text.Sub`.
      */
     public Abbreviated(final Text text, final int max) {
         super(
-            new TextOf(
-                () -> {
-                    final Text abbreviated;
-                    if (text.asString().length() <= max) {
-                        abbreviated = text;
-                    } else {
-                        abbreviated = new FormattedText(
-                            "%s%s",
-                            new Sub(
-                                text,
-                                0,
-                                max - Abbreviated.ELLIPSES.length()
-                            ).asString(),
-                            Abbreviated.ELLIPSES
-                        );
-                    }
-                    return abbreviated.asString();
-                }
+            new Flattened(
+                new Ternary<>(
+                    new ScalarOf<>(() -> new Sticky(text)),
+                    (Text t) -> t.asString().length() <= max,
```

Comment:
@victornoel `LengthOf` could be used here.

Reply:
@andreoss done, thx